### PR TITLE
Add No Internet Bar and Update Plugin Management

### PR DESCRIPTION
**Pull request by akaplo**
Checks the 2nd box in #304.  Adds a global "offline?" object and two global listeners for the [HTML5 online/offline events](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine).  After doing some testing, I noticed that this could be flaky in WebViews (when run natively on devices), so I've also added the [Cordova Network Information Plugin](https://github.com/apache/cordova-plugin-network-information), which exposes the same events.

In other news, I've finally added all plugins to `config.xml`, removed all plugins from git, and updated the gitignore so that the `json` config files stay in git, but not the plugin files themselves.

To get the plugins at their correct versions after this PR is merged, run `cordova prepare`.

For your viewing pleasure, here's a shoddy gif of a network change in action on the native version of the app on a Nexus 5X running Android 7.1.1.

merp

### CODE REVIEW

**Comment by akaplo** [comment]
@petermarathas or @bgregg can one of you guys review?  You can ignore 430 of the 442 files that were changed - just removing plugins from git.

File: www/app.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ angular.module('pvta', ['ionic', 'ngCordova', 'pvta.controllers', 'angularMoment
       cordova.plugins.Keyboard.hideKeyboardAccessoryBar(true);
       cordova.plugins.Keyboard.disableScroll(true);
     }
+    // Set the global variable for whether the device is "online".
```
Comment (petermarathas) [inline www/app.js:25]: 👍

**Review by petermarathas — APPROVED** [review]
Simple and runs smoothly 👍  nice work

**Comment by akaplo** [comment]
Thanks!

**Pull request merged**